### DISCUSSION TITLE
Channel name addition to plot_response_curves()

### DISCUSSION THREAD

**Discussion by FedericoRaimondi** [google/lightweight_mmm#20] [Ideas]
Not sure if this is already in your plans, but I'm adding it here as a feature request:
In `plot_response_curves()` we miss the argument to add the channel name, that would be really nice from a visualization point of view.

Thanks for your effort!

**Comment by pabloduque0** [comment]
Hey @FedericoRaimondi ! Thanks for opening up this one.

Channel names are given at `fit` time to the model as `media_names`. [See the docs here](https://lightweight-mmm.readthedocs.io/en/latest/api.html#lightweight_mmm.lightweight_mmm.LightweightMMM.fit). Then response curve plots should be able to pick it up.

This is given that the same model object can be trained multiple times with different data and so it is consistent with the model for any plot (I know other plots do allow to pass the names). As the model object serves as a center piece for all the library.

We need to see if we allow all plots to independently pass the names or if we stick with what the model object contains and remove it from the plots that allow it. Or just have both options, maybe the latter seems like the easiest for most people to find it for their usage.

**Comment by FedericoRaimondi** [comment]
thanks, indeed it worked.

yes, I missed that as I was thinking to the other plots.
I agree with the 2 options for consistency. Probably to include one time in `fit` could be the best from a usability pov.

**Reply by pabloduque0** [reply]
Yes I agree! I think we also have to make it more visible through examples so users know its there (the `fit` param).

**Comment by AruunnAnanda** [comment]
Hi @pabloduque0,

Can we also add a point on the curve to indicate what our mean spending is to make it more insightful? and we know whether we are spending close saturation levels visually?

Like the one below. 

<img width="939" alt="image" src="https://user-images.githubusercontent.com/107944288/207870130-2490665b-d302-413b-986a-3a1649a5f13a.png">

I don't know if this is something that is already planned. Thanks!

**Reply by pabloduque0** [reply]
Its already available! The idea is that you will run the optimization first and then you can plug in through `optimal_allocation_per_timeunit` when doing that it will also plot the average historic spend.

See [documentation here](https://lightweight-mmm.readthedocs.io/en/latest/api.html#lightweight_mmm.plot.plot_response_curves).

**Reply by Aanai** [reply]
Hello @pabloduque0 

Something strange is going on when I use `optimal_allocation_per_timeunit` while plotting the response curves. The optimal budget (the number of time periods used for optimisation was 4 weeks, but does this make a difference?) is orders of magnitude greater than the average spend for all media channels in the response curve plots. But the optimisation results themselves makes sense (for example: avg_spend for this channel is ~27000, solution.x for Bing Ads is 132036 but from the plot it is ~3.4*10^7) 

![Bing](https://github.com/google/lightweight_mmm/assets/36119600/c0ccf7f5-4937-4d94-b924-d9266d44caad)

An additional question regarding the response curves from lightweightmmm is what "timeframe" do they capture? Let's say the model was built with 2 years historical data and for the Bing Ads channel the total spend during the 2 years was $8 million dollars, does that mean the range in the x-axis for the Bing Ads response curve would be $0 - $8 million? Or is the response curve to be interpreted for each spending cycle (say, one week for a weekly data model). As a channel keeps running, are we "resetting" the adstock or does it keep adding up until every channel eventually saturates?

PS: This is a hill_adstock model and model converged.

**Reply by Aanai** [reply]
I got the answer to the first question. 
optimal_allocation_per_timeunit = media_scaler.transform(optimal_budget_allocation)/n_time_periods 
solves the issue. It would be helpful to mention this in the documentation. 

Would appreciate clarity on how these "forward predict" response curves are plotted. @pabloduque0
